packages/svelte/src/internal/client/types.d.ts

```diff
@@ -264,25 +153,43 @@ export type Block =
 	| DynamicElementBlock
 	| DynamicComponentBlock
 	| HeadBlock
-	| KeyBlock
 	| EachBlock
 	| EachItemBlock
 	| SnippetBlock;
 
+export interface TransitionManager {
+	is_global: boolean;
+	in: () => void;
+	out: (callback?: () => void) => void;
+	stop: () => void;
+}
+
+export interface AnimationManager {
+	measure: () => void;
+	apply: () => void;
+}
+
+export interface Animation {
+	abort: () => void;
+	neuter: () => void;
+	reset: () => void;
+	p: (now: number) => number;
+}
```

Comment:
this could be a good place to add jsdoc for all the different animiation/transition methods, briefly explaining what they do and why they are important. Then the docs in `transitions.js` probably don't need as much content.